### PR TITLE
feat: add code samples and tests for jobs with captions

### PR DESCRIPTION
**Pull request by irataxy**
Fixes b:210892114

https://cloud.google.com/transcoder/images/caption.srt needs to be uploaded to the sample project bucket (cloud-samples-data) under the media/ directory.

### CODE REVIEW

File: media/transcoder/create_job_with_embedded_captions.go [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+	defer client.Close()
+
+	req := &transcoderpb.CreateJobRequest{
+		Parent: fmt.Sprintf("projects/%s/locations/%s", projectID, location),
+		Job: &transcoderpb.Job{
+			OutputUri: outputURI,
+			JobConfig: &transcoderpb.Job_Config{
+				Config: &transcoderpb.JobConfig{
+					Inputs: []*transcoderpb.Input{
+						&transcoderpb.Input{
```
Comment (codyoss) [inline media/transcoder/create_job_with_embedded_captions.go:51]: When instantiating a slice item you can omit the types of the individual items. So here you can remove the `&transcoderpb.Input` part. There are about a dozen of these in this file.

Comment (codyoss) [reply media/transcoder/create_job_with_embedded_captions.go:51]: Also in other files.

**Review by codyoss — CHANGES_REQUESTED** [review]
Just a couple of style nits

File: media/transcoder/create_job_with_embedded_captions.go (hunk 1 above)
Comment (irataxy) [reply media/transcoder/create_job_with_embedded_captions.go]: Done in these files. Filed b/218557754 to clean up the rest of the files.

**Review by codyoss — APPROVED** [review]
Thanks!

**Pull request merged**